daemon/cmd/endpoint.go

```diff
@@ -219,24 +219,31 @@ func (d *Daemon) fetchK8sMetadataForEndpointFromPod(p *slim_corev1.Pod) (*endpoi
 }
 
 type cachedEndpointMetadataFetcher struct {
-	k8sWatcher *watchers.K8sWatcher
+	db         *statedb.DB
+	namespaces statedb.Table[daemonk8s.Namespace]
+	pods       statedb.Table[daemonk8s.LocalPod]
 }
 
-func (cemf *cachedEndpointMetadataFetcher) FetchNamespace(nsName string) (*slim_corev1.Namespace, error) {
-	// If network policies are disabled, labels are not needed, the namespace
-	// watcher is not running, and a namespace containing only the name is returned.
-	if !option.NetworkPolicyEnabled(option.Config) {
-		return &slim_corev1.Namespace{
-			ObjectMeta: slim_metav1.ObjectMeta{
-				Name: nsName,
-			},
-		}, nil
-	}
-	return cemf.k8sWatcher.GetCachedNamespace(nsName)
+func (cemf *cachedEndpointMetadataFetcher) FetchNamespace(nsName string) (ns daemonk8s.Namespace, found bool) {
```

Comment:
AFAIU we are changing the way we handle `option.NetworkPolicyEnabled(...)`. Before that option was gating the enabling of the watcher in `allResourceGroups`:

```go
	if option.NetworkPolicyEnabled(option.Config) {
		// Namespaces can contain labels which are essential for
		// endpoints being restored to have the right identity.
		// Namespaces are only used when network policies are enabled.
		k8sGroups = append(k8sGroups, resources.K8sAPIGroupNamespaceV1Core)
	}
```
And in case np was not enabled, here we returned a namespace obj without any label.

Shouldn't we keep that logic with the reflector too (like disabling the reflector if np are disabled)?

Reply:
I'm not sure what you meant by "disabling the reflector" as I didn't see any code like that. The above was just about "caches synced" thing which is now implemented in the endpoint metadata fetcher (by waiting for table to initialize). I'll keep the logic of returning nil labels if network policy is enabled for endpoints.